docs/reference/builder.md

```diff
@@ -1177,6 +1182,16 @@ the variable's value in the `ENV` references the `ARG` variable and that
 variable is changed through the command line. In this example, the `ENV`
 command causes the image to include the value.
 
+If an ENV instruction overrides an ARG instruction of the same name, like this Dockerfile:
+
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 ENV CONT_IMG_VER hello
+4 RUN echo $CONT_IMG_VER
+```
+There will be no cache miss on line 3.
```

Comment:
perhaps:
```
There will be no cache miss on line 3 because
the value of `CONT_IMG_VER` is now a constant
(`hello`) and therefore the list of environment variables,
and values, used on the `RUN` will not change between builds.
```